Date select filter

carnage opened the pull request:
This pull request refactors the callback filters used in date and date time select form elements into concrete filters. This allows the filter to be reapplied when the user is using annotations and sets prefer_form_input_filter to true (as is done by default by Doctrine's form annotation builder) fixing a bug with validation.

As a bonus, the filter also contains flags to return null if the data select is empty (currently, empty selects return a string of -- or -- ::) The Date validator has also been updated with a flag to allow for this new behaviour.

ameir committed:
Add ability to set SMTP envelope addresses

ameir committed:
Implement code changes, per @Ocramius' suggestions

carnage committed:
Added date select filter and updated element to use it

carnage committed:
Added date time select filter and updated element to use it

carnage committed:
Added new filters to filter manager

carnage committed:
Added licence

carnage committed:
Abstracted date select filter logic to reuse for month select

carnage committed:
Added unit test for month select filter

carnage committed:
Added month select filter to manager and element

carnage committed:
Updated doc blocks

turrsis committed:
fix restore_error_handler();

Martin-P committed:
Fixes session validators

samsonasik committed:
added Mime\Exception\InvalidArgumentException instead

samsonasik committed:
fixes cs

samsonasik committed:
fixes @throw docblock

samsonasik committed:
fixes cs : unused use

carnage commented:
Theses filters are designed to be used in conjunction with the (Date|Month|DateTime)Select form elements: given that we plan to split the repos; should these live in a sub-namespace of Form?

The magic constants could then be referenced from the Form elements/view helpers that also use them

samsonasik committed:
update copyright year in InvalidArgumentException

Victory committed:
A failing test to show that FormButtonHelper does not translate $buttonContent as a parameter

Victory committed:
A failing test to show that FormButtonHelper does not translate $buttonContent as a parameter

Victory committed:
fix FormButton $buttonContent parameter not being translated

Victory committed:
fix FormButton $buttonContent parameter not being translated

Victory committed:
remove extra if/else in FormButton render

Victory committed:
fix test pattern in FormDateTimeSelectTest::testCanRenderTextDelimiters

Victory committed:
resolve merge conflict for FormButton

Victory committed:
Merge branch 'form-button-translate-content-param' of github.com:Victory/zf2 into form-button-translate-content-param

Conflicts:
	library/Zend/Form/View/Helper/FormButton.php

Commit:
add test for nested glob patterns

Commit:
fix array syntax

Martin-P committed:
Added test for array of connection params

jaapio committed:
fix some of the documentation errors

jaapio committed:
implement incompleet test

jaapio committed:
improve code coverage of DoctrineAnnotationParser

jaapio committed:
improve code coverage of GenericAnnotationsParser

jaapio committed:
add testcase for invalid argument

mwillbanks committed:
Forward port #7200

Merge branch 'bakura10-hotfix-datetime' into develop

jaapio committed:
add test for setFlags

jaapio committed:
add testcase for setDocblock

jaapio committed:
remove unused use statement

marc-mabe committed:
Cache: fix mongo db options

chadicus committed:
Error in FirePhp log writer

chadicus committed:
Add unit test for FirePhpInterface check

Commit:
Added fix to prevent method lines to be intended again and again

Commit:
Added unit test to test Zend\Code\Generator\MethodGenerator::clearBodyIndention() method

Commit:
Added file header to ZendTest\Code\Generator\TestAsset\TestSampleSingleClassMultiLines

samsonasik committed:
replace is_a() with instanceof

Martin-P committed:
Added additional test and fix for that specific test

moderndeveloperllc committed:
Fix for #7219 - Validation failure fr-CH in testValidationFailures due to change in CLDR 26. Just removed that one instance that is actually valid now.

weierophinney committed:
DBA inifile is not supported for caching

We recently noticed failures occuring for the DbaInifile cache storage tests.
The errors reported indicated an inability to perform `dba_replace()`
operations, with no indication of why this was.

On digging through the various DBA documentation on php.net, I discovered that
the `dba_replace` and `dba_insert()` functions are only reliable on GDBM and
QDBM handlers; any other handler may result in unexpected or unintended
behavior.

I have tested against flatfile, GDBM, DB4, and GDBM, and none of these show any
issues; however, inifile reliably has issues at this point. As such, we can no
longer support that version.

This patch raises an exception if the inifile handler is provided as a DBA
option, and the tests have been modified to ensure that the exception is raised.

Commit:
Skip tests if ext-intl is not installed

weierophinney committed:
Merge branch 'security/zf2015-02-2.3' into develop

ZF2015-02 patches

Conflicts:
	library/Zend/Db/Adapter/Platform/Postgresql.php
	tests/ZendTest/Db/Adapter/Platform/PostgresqlTest.php

weierophinney committed:
Merge branch 'releases/2.3.5' into develop

2.3.5 release readiness

Conflicts:
	README.md
	library/Zend/Version/Version.php

weierophinney committed:
Merge branch 'version/bump' into develop

Bump to next dev version

Conflicts:
	README.md
	library/Zend/Version/Version.php

anton-kotik committed:
Remove unnecessary ternary operators

anton-kotik committed:
Fix syntax error

Commit:
Detect https on reversed proxy

Commit:
Left and right outer joins constant names rearanged

Additionally removed unnecessary inner joins constants

weierophinney committed:
Merge pull request #4726 from postalservice14/add_maxlength_to_param_container

Add maxLength to ParameterContainer

weierophinney committed:
Merge branch 'hotfix/4726' into develop

Close #4726

weierophinney committed:
Merge pull request #6194 from ojhaujjwal/hydrator-chain

Created hydrator strategy chain

weierophinney committed:
Merge branch 'feature/6194' into develop

Close #6194

weierophinney committed:
Merge pull request #6246 from rodmcnew/6244-translation-loaders-from-config

 I18n - Allow custom translation loaders to be injected via module config #6244

Conflicts:
	library/Zend/Mvc/Service/ServiceListenerFactory.php

weierophinney committed:
[#6246] CS fixes

- Removed extraneous line

weierophinney committed:
Merge branch 'feature/6246' into develop

Close #6246

weierophinney committed:
Merge pull request #6568 from PhilGrayson/master

Add failing test for encoded routes with a query

weierophinney committed:
Merge branch 'feature/6568' into develop

Close #6568

weierophinney committed:
Merge pull request #6669 from ameir/imap-subscribe

Support subscribing to folders in IMAP

weierophinney committed:
[#6669] CS fixes

- Trailing whitespace

weierophinney committed:
Merge branch 'feature/6669' into develop

Close #6669

ameir committed:
Fix SmtpProtocolSpy and update corresponding tests.

samsonasik committed:
move ext-ldap from suggest to require

ameir committed:
Remove debug line.

ameir committed:
Update Envelope class and add tests.

ameir committed:
Fix tests in Mail/Protocol/SmtpTest.php.

Commit:
Merge branch 'master' of https://github.com/zendframework/zf2

Commit:
Adding string return type for __invoke

Lansoweb committed:
Code cleanup 

Removing unnecessary variable, fixing showResponse call

stefanotorresi committed:
add Zend\Mvc\HttpMethodListener

stefanotorresi committed:
add Zend\Mvc\Service\HttpMethodListenerFactory

samsonasik committed:
Fixes #7238 : Logger::log() exception

stefanotorresi committed:
make Application::bootstrap() ready for HttpMethodListener integration

stefanotorresi committed:
refactor Application::bootstrap() tests

previous tests assertions were too specific and
prevented the introduction of additional default event listeners

Martin-P committed:
Removed non existing curl options

sandrokeil committed:
Introducing NavigationAbstractServiceFactory

improved-broccoli committed:
Add input_filter_specs feature from zfcampus/zf-content-validation and associated tests

Commit:
Remove CURLOPT_POSTFIELDS from set of invalid curl options

Right now there is no way to upload a file without loading the entire file into memory twice.
That is quite a bummer, because CURL supports it just fine. The CURLOPT_POSTFIELDS allows doing
just that.

Now it would of course be exellent if the built in file upload stuff in ZF would just use
said option. But as a workaround, it would be great to be able to override it.

Another thing: Getting this to work properly also requires messing around with the ENC_TYPE
in the client. Basically manually overriding it to ENC_FORMDATA.

improved-broccoli committed:
Change copyright/license headers

Commit:
Test case for setting CURLOPT_POSTFIELDS

weierophinney committed:
Revert "Merge pull request #7034 from marc-mabe/MongoDB_to_MongoDb"

See #7034

This reverts commit e97c3ddcb4263ca111fe8c53ce00b935634c690a, reversing
changes made to 7440246ae32bd3dab5cff7dac5afe17f16dc8bc8.

The changes made in the original merge are backwards incompatible, and should
wait until v3 or Zend\Cache. Renaming at this time would break code for existing
users, particularly those who were referencing the classes by name.

weierophinney committed:
Revert "Revert "Merge pull request #7034 from marc-mabe/MongoDB_to_MongoDb""

See #7034

This reverts commit a1dc4b0883e97a01d89472a5d3eca8e8c55a1e3a.

The renames that happened were in unreleased code. As such, they were 100% safe.
This commit reverts the revert, restoring the original merge.

weierophinney committed:
Merge pull request #6709 from nikolaposa/feature/html_tag_helper

HtmlTag helper

weierophinney committed:
Merge branch 'feature/6709' into develop

Close #6709

weierophinney committed:
[#6709] Updated copyright

- to 2015

weierophinney committed:
Merge branch 'feature/6709' into develop

Update copyright in new files introduced in #6709

weierophinney committed:
Merge pull request #6712 from carnage/annotation-factory

Added form annotation builder factory

Conflicts:
	library/Zend/Mvc/Service/ServiceListenerFactory.php

weierophinney committed:
[#6712] Updated copyright year to 2015

weierophinney committed:
Merge branch 'feature/6712' into develop

Close #6712

weierophinney commented on library/Zend/Filter/AbstractDateDropdown.php:
```diff
@@ -0,0 +1,107 @@
+            $value = vsprintf($this->format, $value);
+        }
+
+        return $value;
```
Maybe create a test covering this, demonstrating a bad format and/or value, and an expected exception? That may help you or others to come up with a solution.

weierophinney commented on library/Zend/Filter/DateTimeSelect.php:
```diff
@@ -0,0 +1,68 @@
+    {
+        if (is_array($value)) {
+            if ($this->nullOnEmpty() && (
+                    empty($value['year']) ||
```
@Ocramius confirmed - these keys are all defined as part of the date input type in HTML5.

weierophinney commented:
@carnage I'd like to merge, but:

- We need a rebase
- There are open questions about how to handle invalid inputs

If you can resolve these in the next two weeks, I can keep it in the 2.4 release.

weierophinney committed:
Merge pull request #6731 from bgotink/dev-filefilessize

fix for FilesSize validator usage with Input

weierophinney committed:
Merge branch 'feature/6731' into develop

Close #6731

weierophinney committed:
Merge pull request #6746 from nickpeirson/DI-RuntimeCompiler-Optimisation

Di runtime definition optimisation - Cache result of processing and inline the check

Conflicts:
	library/Zend/Di/Definition/RuntimeDefinition.php

weierophinney committed:
Merge branch 'feature/6746' into develop

Close #6746

weierophinney committed:
Merge pull request #6747 from nickpeirson/DI-DefinitionList-Optimisation

DI definition list optimisation - Cache classes upfront where possible to reduce iterations

weierophinney committed:
[#6747] Update to CS for consistency

- space around operators and between parens/opening braces
- use strict comparison when checking return value of assignment in a
  conditional

weierophinney committed:
Merge branch 'feature/6747' into develop

Close #6747

weierophinney committed:
Merge pull request #6752 from carnage/label-position-option

Added ability to specify label position as an element label option

Merge fixes test errors introduced in the original patch.

Conflicts:
	library/Zend/Form/View/Helper/FormRow.php

weierophinney committed:
Merge branch 'feature/6752' into develop

Close #6752

weierophinney committed:
Merge pull request #6754 from carnage/fix-annotation-builder-ordering

Added preserve_defined_order flag to annotation builder

weierophinney committed:
Merge branch 'feature/6754' into develop

Close #6754

coolms commented on library/Zend/Filter/AbstractDateDropdown.php:
```diff
@@ -0,0 +1,107 @@
+     * @var string
+     */
+    protected $format = '';
+
```
A constructor should be here, otherwise there is no way setting options to filter. As an example:

```php
/**
 * Constructor
 *
 * @param array|Traversable $options OPTIONAL
 */
public function __construct($options = null)
{
    if (static::isOptions($options)) {
        $this->setOptions($options);
    }
}
```

coolms commented on library/Zend/Validator/Date.php:
```diff
@@ -101,6 +107,24 @@ public function setFormat($format = self::FORMAT_DEFAULT)
+    {
+        return $this->allowNull;
+    }
+
```
The name of this method isn't correct. It should be getAllowNull() because of the way options are set into validator. \Zend\Validator\AbstractValidator::setOptions($options) tries to call AllowNull() method first with option's value as parameter, whereupon option will not be set.

weierophinney replied:
@coolms I personally disagree, as I think the proposed syntax reads better in conditionals:

```php
if ($date->allowNull()) {
    // ...
}
```

Since it's a method providing a boolean response, and will be used for conditionals, the focus should not be on whether it's _getting_ a value, but on what it returns.

Commit:
Add support for non 443 port

weierophinney committed:
Merge pull request #7235 from Lansoweb/master

Fixing Docblock return type for __invoke

weierophinney committed:
Merge branch 'feature/7235' into develop

Close #7235

weierophinney committed:
Merge pull request #7239 from samsonasik/hotfix/logger-exception

Fixes #7238 : Logger::log() exception message

Conflicts:
	tests/ZendTest/Log/LoggerTest.php

weierophinney committed:
Merge branch 'feature/7239' into develop

Close #7239

weierophinney committed:
Merge pull request #7241 from Martin-P/hotfix/curl-adapter

Removed non existing curl options

weierophinney committed:
Merge branch 'feature/7241' into develop

Close #7241

weierophinney committed:
Merge pull request #7245 from sandrokeil/navigation-abstract-service-factory

[Navigation] Introducing NavigationAbstractServiceFactory

weierophinney committed:
Merge branch 'feature/7245' into develop

Close #7245

weierophinney committed:
Merge pull request #7248 from researchgate/allow-curlopt-postfields

Remove CURLOPT_POSTFIELDS from set of invalid curl options

weierophinney committed:
Merge branch 'feature/7248' into develop

Close #7248

davidwindell committed:
Remove speech marks

weierophinney committed:
Merge pull request #6783 from carnage/continue-if-empty-annotation

Added continueIfEmpty annotation to Zend Form

weierophinney committed:
[#6783] Incorporate feedback

- Update copyright date
- Simplify original assigment of flag

weierophinney committed:
Merge branch 'feature/6783' into develop

Close #6783

weierophinney committed:
Added StandaloneExtensionManager as a drop-in extension plugin resolver

Essentially mimics the AbstractPluginManager, but does so as a
standalone implementation of ExtensionManagerInterface.

weierophinney committed:
Reader should use StandaloneExtensionManager by default

Modified getExtensionManager to lazyload a StandaloneExtensionManager
instead of ExtensionManager. This required an update to a test that was
testing integration of registerExtension with the EM instance itself.

weierophinney committed:
CS fix noticed when running php-cs-fixer

TRUE vs true

weierophinney committed:
Merge pull request #6267 from davidwindell/hotfix-address-commas

Support commas in names

weierophinney committed:
Merge branch 'feature/6267' into develop

Close #6267

weierophinney committed:
Merge pull request #6756 from samsonasik/feature/implement-todo-mime-part

implements @todo for setters/getters at Zend\Mime\Part

weierophinney committed:
Make syntax consistent with the framework

- We're migrating to `!` followed by a space
- If exceptions need interpolation, we use sprintf

weierophinney committed:
Merge branch 'feature/6756' into develop

Close #6756

weierophinney committed:
Merge pull request #6758 from samsonasik/imp.todo.adapter

add getAdapterInstance() that return adapter instance at Zend\Filter\Encrypt

weierophinney committed:
Adapt to be consistent with framework

- We're migrating to `!` followed by a space
- Use sprintf() when formatting exception messages that use
  interpolation
- Use magic constants where possible to reduce hard-coding

weierophinney committed:
Merge branch 'feature/6758' into develop

Close #6758

weierophinney committed:
Merge pull request #6790 from turrsis/hotfix/db-adapter-platform/sqlServerTest

Test/Db/Adapter/Platform/SqlServerTest restore_error_handler();

weierophinney committed:
Fix reported CS errors

- removed space before closing paren

weierophinney committed:
Merge branch 'feature/6790' into develop

Close #6790

improved-broccoli committed:
Disable input_filter_specs feature by default. Update tests.

weierophinney committed:
Merge pull request #6799 from Pittiplatsch/db_predicate_nextcombination

Zend\Db: Allow adding native predicates to sql queries, respecting nextPredicateCombineOrder

weierophinney committed:
Merge branch 'feature/6799' into develop

Close #6799

weierophinney committed:
Merge pull request #6827 from hissy/no-plural-form-lang

Fix inability to translate to languages which don't have plural forms

weierophinney committed:
Merge branch 'feature/6827' into develop

Close #6827

weierophinney committed:
Merge pull request #6856 from jameswithers/patch-1

Soap Connection Timeout

weierophinney committed:
Merge branch 'feature/6856' into develop

Close #6856

weierophinney committed:
Merge pull request #6931 from AGmakonts/patch-1

Added consts for left and right outer joins

weierophinney committed:
Merge branch 'feature/6931' into develop

Close #6931

weierophinney committed:
Merge pull request #6962 from gabrielsomoza/gabriel/zf-6960-whitelist-filter

New Whitelist / Blacklist Filter

weierophinney committed:
Updated copyright year

weierophinney committed:
Merge branch 'feature/6962' into develop

Close #6962

weierophinney committed:
Merge pull request #7016 from raum42/master

Adapt Json\Json for better prettyPrint support

weierophinney committed:
Update per php-cs-fixer

- Removed unneeded line following brace

weierophinney committed:
Merge branch 'feature/7016' into develop

Close #7016

weierophinney committed:
Merge pull request #7091 from Lansoweb/develop

Added Checkbox Console Prompt and its tests

weierophinney committed:
Updates recommended by php-cs-fixer

weierophinney committed:
Merge branch 'feature/7091' into develop

Close #7091

weierophinney committed:
Merge pull request #7102 from neilime/patch-3

Add Kosovo phone number specifications

weierophinney committed:
Merge branch 'feature/7102' into develop

Close #7102

weierophinney committed:
Merge pull request #7104 from veewee/feature-bytes-filter

DataUnitFormatter filter

weierophinney committed:
Merge branch 'feature/7104' into develop

Close #7104

weierophinney committed:
Merge pull request #7120 from RalfEggert/feature/value-generator-array-comma

Added trailing comma for multiline arrays

weierophinney committed:
Merge branch 'feature/7120' into develop

Close #7120

weierophinney committed:
Merge pull request #7121 from ins0/fix/curl-ssl-verify

Feature: Added sslverifypeer to curl adapter

weierophinney committed:
Merge branch 'feature/7121' into develop

Close #7121

weierophinney committed:
Merge pull request #7122 from tklever/feature-paginator-iterator

Paginator Iterator

weierophinney committed:
Merge branch 'feature/7122' into develop

Close #7122

weierophinney committed:
Merge pull request #7125 from wesleywillians/zf2-7124

Adding ImportInterface for Zend\Feed\Reader\Reader

weierophinney committed:
Merge branch 'feature/7125' into develop

Close #7125
Fixes #7124

weierophinney committed:
Merge pull request #7128 from Martin-P/hotfix/session-validators

SessionManagerFactory + session validators via config makes validation always fail

weierophinney committed:
Fix CS issues reported by php-cs-fixer

weierophinney committed:
Merge branch 'feature/7128' into develop

Close #7128

weierophinney committed:
Merge pull request #7152 from kusmierz/feature/4936-key-fail

Fixes #4936

weierophinney committed:
Merge branch 'feature/7152' into develop

Close #7152
Fixes #4936

weierophinney committed:
Merge pull request #7159 from Martin-P/hotfix/abstract-unicode-filters

Use getEncoding() instead of accessing options array directly

weierophinney committed:
Merge branch 'feature/7159' into develop

Close #7159
Fixes #7147

Logiqu2 committed:
fixed count error in config class

weierophinney committed:
Added file-level docblock.

weierophinney committed:
Merge pull request #7256 from Logiqu/config-count-fix

fixed count error in config class

weierophinney committed:
Merge branch 'feature/7256' into develop

Close #7256

weierophinney committed:
Merge pull request #7067 from marc-mabe/travis/cache

Travis: running more tests for cache component

Conflicts:
	.travis.yml

weierophinney committed:
Merge branch 'feature/7067' into develop

Close #7067

weierophinney committed:
Merge pull request #7163 from prolic/patch-10

Fix typos in inArray validators phpdoc

weierophinney committed:
Merge branch 'feature/7163' into develop

Close #7163

weierophinney committed:
Merge pull request #7177 from MarcelTO/master

Oracle quoting fix for values with single quotes

weierophinney committed:
Merge branch 'feature/7177' into develop

Close #7177

weierophinney committed:
Merge pull request #7178 from samsonasik/hotfix/ldap-composer

Fixes #7136 : add ext-ldap as required extension in Ldap\composer.json

weierophinney committed:
Merge branch 'feature/7178' into develop

Close #7178

weierophinney committed:
Merge pull request #7181 from hver/paginator-cache-interfaces

Remove hidden TaggableInterface dependencies from paginator

weierophinney committed:
Merge branch 'feature/7181' into develop

Close #7181
Fixes #7174

weierophinney committed:
Merge pull request #7182 from samsonasik/copyright-2015

update copyright year to 2015 that still uses 2013 in "develop"

weierophinney committed:
Merge branch 'feature/7182' into develop

Close #7182

weierophinney committed:
Merge pull request #7183 from samsonasik/hotfix/copyright-2015

update copyright year to 2015 that still uses 2013 in "master"

weierophinney committed:
Merge branch 'feature/7183' into develop

Close #7183

weierophinney committed:
Merge pull request #7184 from Victory/form-button-translate-content-param

FormButton $buttonContent parameter is now translated

weierophinney committed:
Merge branch 'feature/7184' into develop

Close #7184

weierophinney committed:
Merge pull request #7188 from Victory/fix-test-can-render-text-delimiters

fix test pattern in FormDateTimeSelectTest::testCanRenderTextDelimiters

weierophinney committed:
Merge branch 'feature/7188' into develop

Close #7188

weierophinney committed:
Merge pull request #7189 from stefanotorresi/test-nested-glob-patterns

add test for nested glob patterns

weierophinney committed:
Merge branch 'feature/7189' into develop

Close #7189

weierophinney committed:
Merge pull request #7190 from Martin-P/hotfix/pdo-connection-dsn-port-test

Test for array of PDO connection params

weierophinney committed:
Merge branch 'feature/7190' into develop

Close #7190

weierophinney committed:
Merge pull request #7194 from jaapio/docblock_fixes

fix some of the documentation errors

weierophinney committed:
Merge branch 'feature/7194' into develop

Close #7194

weierophinney committed:
Merge pull request #7201 from jaapio/cleanup/zfcode

Additional tests for zf-code

weierophinney committed:
Fix errors reported by php-cs-fixer

weierophinney committed:
Merge branch 'feature/7201' into develop

Close #7201

weierophinney committed:
Merge pull request #7203 from marc-mabe/cache/fixMongoOptions

Cache: fix mongo options

weierophinney committed:
Perform argument validation

Tests indicated that an InvalidArgumentException should be thrown if the
`$resource` argument to `setResource` was not an array or
`MongoCollection`; however, no exception was thrown.

weierophinney committed:
Fix errors reported by php-cs-fixer

- Trailing whitespace

weierophinney committed:
Merge branch 'feature/7203' into develop

Close #7203

weierophinney committed:
Merge pull request #7210 from RalfEggert/bug/method-indention-from-class-reflection

Added fix to prevent method lines to be intended again and again

weierophinney committed:
Address php-cs-fixer errors

- Remove trailing line between closing braces.

weierophinney committed:
Merge branch 'feature/7210' into develop

Close #7210
Fixes #7209

weierophinney committed:
Merge pull request #7213 from chadicus/master

Error in FirePhp log writer

weierophinney committed:
Merge branch 'feature/7213' into develop

Close #7213
Fixes #7202

weierophinney committed:
Merge pull request #7215 from samsonasik/is_a

replace is_a() with instanceof

weierophinney committed:
Merge branch 'feature/7215' into develop

Close #7215

weierophinney committed:
Merge pull request #7225 from moderndeveloperllc/hotfix/FloatTest-testValidationFailures-fix

Fix for #7219 - Validation failure fr-CH in testValidationFailures

weierophinney committed:
Merge branch 'feature/7225' into develop

Close #7225

weierophinney committed:
Merge pull request #7231 from gianarb/feature/fix-zendtest-view-ext-intl

Skip tests in ZendTest\View if ext-intl is not installed

weierophinney committed:
Merge branch 'feature/7231' into develop

Close #7231

weierophinney committed:
Merge pull request #7232 from 27cm/iif

Remove unnecessary ternary operators

Conflicts:
	library/Zend/Db/Adapter/Driver/IbmDb2/Connection.php

weierophinney committed:
Merge branch 'feature/7232' into develop

Close #7232

weierophinney committed:
Merge pull request #7233 from rstgroup/master

Detect https on reversed proxy

weierophinney committed:
Merge branch 'feature/7233' into develop

Close #7233

weierophinney committed:
Merge pull request #7240 from stefanotorresi/feature/http-method-listener

Feature: Zend\Mvc\HttpMethodListener

weierophinney committed:
Address issues raised by php-cs-fixer

- single line after imports

weierophinney committed:
Merge branch 'feature/7240' into develop

Close #7240
Fixes #6385

weierophinney committed:
Merge pull request #7247 from jbenoit2011/import-input_filter_specs-from-zfcampus/zf-content-validation

Add input_filter_specs feature from zfcampus/zf-content-validation and associated tests

weierophinney committed:
Use specific return types when possible

weierophinney committed:
Spaces between @param and @return annotations are not necessary

weierophinney committed:
Merge branch 'feature/7247' into develop

Close #7247

weierophinney committed:
Merge pull request #6570 from ameir/smtp-envelope

Add ability to set SMTP envelope addresses

weierophinney committed:
Incorporate feedback

- Remove fluent interfaces
- Ensure docblocks are up-to-date
- Fix CS issues

weierophinney committed:
Update tests to remove fluent usage.

weierophinney committed:
Merge branch 'feature/6570' into develop

Close #6570

weierophinney committed:
Merge pull request #7056 from hschletz/inputfilter-refactor

Refactoring and bugfixes for InputFilter component

Conflicts:
	tests/ZendTest/InputFilter/BaseInputFilterTest.php

weierophinney committed:
Address issues raised by php-cs-fixer

- Anonymous function definitions

weierophinney committed:
Merge branch 'feature/7056' into develop

Close #7056

snapshotpl committed:
Clean reason phrase when status code is changed

snapshotpl committed:
Test comment

ezimuel committed:
Merge branch 'feature/6419' of https://github.com/weierophinney/zf2 into weierophinney-feature/6419

ezimuel committed:
Merge branch 'weierophinney-feature/6419' into develop

weierophinney committed:
Use 5.3 array notation for compatibility

weierophinney committed:
Merge pull request #7259 from snapshotpl/feature/http-phrase

Override reason phrase by changing status code

weierophinney committed:
Merge branch 'feature/7259' into develop

Close #7259

weierophinney committed:
Merge branch 'hotfix/array-notation' into develop

Close #7265

ezimuel committed:
Merge branch 'weierophinney-hotfix/cache-dba-inifile' into develop

weierophinney committed:
Correct errors reported by php-cs-fixer

- in php-cs-fixer 1.5 on PHP 5.6

weierophinney committed:
Correct errors reported by php-cs-fixer

- in php-cs-fixer 1.5 on PHP 5.6

weierophinney committed:
Merge branch 'hotfix/phpcs-5.6' into develop

Close #7268

carnage committed:
Added tests for invalid input values

carnage committed:
Added code to throw exception if filtering of an input is impossible

carnage commented:
Rebase messed this up. Reopened against develop
